pkgs/applications/audio/carla/default.nix

```diff
@@ -67,6 +67,11 @@ stdenv.mkDerivation rec {
         --prefix PATH : "$program_PATH:${which}/bin" \
         --set PYTHONNOUSERSITE true
     done
+
+    # Hardcode --with-appname argument to `carla` since python wrapper is being wrapped again by QT wrapper.
```

Comment:
Can we just avoid the double wrapping by giving the python wrapper the qtwrapper args?

Reply:
could we? I think both wrappers are package set specific. What I think we could do to cat them into a one bash script.

Reply:
Are we sure that the file `.carla-wrapper` is not coming from Carla directly? I can't find any reference to `--with-appname` in the nixpkgs repository, but [this file](https://github.com/falkTX/Carla/blob/main/data/carla) from carla's repo is nearly exactly our `.carla-wrapper` file. I'm the feeling that the python wrapper only adds some text on top of the `.py` files, and therefore the qtwrapper is the only function that actually create a new file and move the old carla file. I may be wrong, but if you have more information I would be curious to know.

Reply:
@tobiasBora you're right. Seems I was originally also right under the issue. The whole --with-appname business is part of carla. Anyway what we could do instead of patching this value is to just cat the content generated by qt wrapper - that way we would avoid having any wrappers. But I somehow think that layering the wrappers might be still a bit more maitainable going forward than trying to generate some unique file in unique way here. What I find a bit odd is that carla is using `$0` in it's own code while really supporting only one possible value there. That seems to be a bit confusing to me.

Reply:
@tobiasBora Do you by any chance know why carla does that? I think we could also suggest the change upstream which would be the nicest solution for this. Given the value other than `carla` is problematic there I think it could save troubles beyond just nixpkgs.

Reply:
I guess that what is important for carla is that `$0` must be the name of an executable file available in your path that points to carla. So if a user wants to rename the file to `mycarla` instead, the code would still work with `$0` while it would be broken with `carla`... and carla did not expect nix to kill this feature.

Concerning the copy of the wrapper code inside carla file directly, I'd guess it will be harder to maintain indeed... It may be a good idea to include wrappers directly inside the `bash` script when possible, instead of adding a level of wrapping, but I guess it should be done by the wrapper code directly, not manually... So it may be an independent problem to raise to Nix community.

Reply:
I suggest you use `buildPythonApplication` and set
```
preFixup = ''
  makeWrapperArgs+=("''${qtWrapperArgs[@]}")
'';
```

Reply:
I can look more into this later. But also if you want to do that feel free to open another PR super-seeding this one.